lua/telescope/previewers/buffer_previewer.lua

```diff
@@ -872,4 +872,39 @@ previewers.display_content = defaulter(function(_)
   }
 end, {})
 
+previewers.buffers = defaulter(function(opts)
+  opts = opts or {}
+  local cwd = opts.cwd or vim.loop.cwd()
+  return Previewer:new {
+    title = function()
+      return "Buffers"
+    end,
+    dyn_title = function(_, entry)
+      return Path:new(from_entry.path(entry, true)):normalize(cwd)
+    end,
+    preview_fn = function(self, entry, status)
+      if vim.api.nvim_buf_is_valid(entry.bufnr) then
+        vim.api.nvim_win_set_buf(status.preview_win, entry.bufnr)
```

Comment:
I think you need to make sure that buffer is loaded. So `if vim.api.nvim_buf_loaded ==false then vim.fn.bufload` or so

For example if you load a session, all buffers are unloaded, so you will end up with no highlighting or something like that.

Reply:
Okay just tested we dont need to. In the past in ran into this but maybe that was bug upstrem and `nvim_win_set_buf` now does ensure that. So better ignore that comment here :rofl: